tonic/src/codec/decode.rs

```diff
@@ -393,8 +393,8 @@ impl<T> Stream for Streaming<T> {
 
     fn poll_next(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Option<Self::Item>> {
         loop {
-            if let State::Error(status) = &self.inner.state {
-                return Poll::Ready(Some(Err(status.clone())));
+            if let State::Error(status) = &mut self.inner.state {
+                return Poll::Ready(status.take().map(Err));
```

Comment:
Maybe add a short comment here on what the desired behavior is?

Reply:
of course!